src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -51,19 +66,31 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
         // TokenPermissions should have been set up in the request by TokenPermissionsInterceptor
         final var tokenPermissions = getTokenPermissions(request)
                 .orElseThrow(() -> new IllegalStateException("UserAuthenticationInterceptor - TokenPermissions object not present in request"));
-
-        // Check the user has the company_incorporation=create permission
-        boolean hasCompanyIncorporationCreatePermission = tokenPermissions.hasPermission(Key.COMPANY_INCORPORATION, Value.CREATE);
+        String companyNumberInScope = getCompanyNumberInScope(request);
+        boolean isRoeUpdateJourneyRequest = isRequestRoeUpdate(companyNumberInScope);
 
         var authInfoMap = new HashMap<String, Object>();
         authInfoMap.put(TRANSACTION_ID_KEY, transactionId);
         authInfoMap.put("request_method", request.getMethod());
-        authInfoMap.put("has_company_incorporation_create_permission", hasCompanyIncorporationCreatePermission);
+        authInfoMap.put("company_number_in_scope", companyNumberInScope);
+        authInfoMap.put("is_roe_update_journey_request", isRoeUpdateJourneyRequest);
 
-        if (hasCompanyIncorporationCreatePermission) {
-            ApiLogger.debugContext(reqId, "UserAuthenticationInterceptor authorised with company_incorporation=create permission",
-                    authInfoMap);
-            return true;
+        if (isRoeUpdateJourneyRequest) {
+            // Check the user has the company_oe_annual_update=create permission for ROE Update
+            boolean hasCompanyOeAnnualUpdateCreatePermission = tokenPermissions.hasPermission(Key.COMPANY_OE_ANNUAL_UPDATE, Value.CREATE);
+            authInfoMap.put("has_company_oe_annual_update_create_permission", hasCompanyOeAnnualUpdateCreatePermission);
+            if (hasCompanyOeAnnualUpdateCreatePermission) {
+                return doRoeUpdateJourneyChecks(request, response, reqId, companyNumberInScope, transactionId, authInfoMap);
+            }
+        } else if (StringUtils.isEmpty(companyNumberInScope)){
```

Comment:
I'm wondering why the `else if` was added? We didn't need this originally when just doing the check for registration, so if not on the update journey then we should apply logic as before (I'd have thought).

Reply:
Yeah looks like with the refactor we need this check now ....

Reply:
This has now been fixed -- have reworked the logic .....